### PR TITLE
imgcrypt: init at 1.1.1

### PR DESCRIPTION
**Pull request by mikroskeem**
###### Motivation for this change

Package imgcrypt to add support for encrypted container images for e.g containerd and CRI-o

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/imgcrypt/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  ];
+
+  vendorSha256 = null;
+  subPackages = [ "cmd/ctd-decoder" "cmd/ctr" ];
+
+  postFixup = ''
+    mv $out/bin/ctr $out/bin/ctr-enc
+  '';
+
+  doCheck = true;
```
Comment (fabaff) [inline pkgs/applications/virtualization/imgcrypt/default.nix:24]: Is a default, can be removed.

File: pkgs/applications/virtualization/imgcrypt/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+buildGoModule rec {
```
Comment (fabaff) [inline pkgs/applications/virtualization/imgcrypt/default.nix:2]: Better readability for humans.

```suggestion
{ buildGoModule, fetchFromGitHub, lib }:

buildGoModule rec {
```

File: pkgs/applications/virtualization/imgcrypt/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  pname = "imgcrypt";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "containerd";
+    repo = "imgcrypt";
```
Comment (fabaff) [inline pkgs/applications/virtualization/imgcrypt/default.nix:8]: Opinionated change

```suggestion
    repo = pname;
```

File: pkgs/applications/virtualization/imgcrypt/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    mv $out/bin/ctr $out/bin/ctr-enc
+  '';
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Image encryption library and command line tool";
+    license = licenses.asl20;
+    platforms = platforms.linux;
+    inherit (src.meta) homepage;
```
Comment (fabaff) [inline pkgs/applications/virtualization/imgcrypt/default.nix:30]: Personally I'm not a huge fan of this pattern as it requires additional effort to get to the upstream location from git.

Comment (mikroskeem) [reply pkgs/applications/virtualization/imgcrypt/default.nix:30]: So you'd simply prefer being able to get the homepage url directly from derivation file? Sounds like a fair change what many would prefer to have.

**Review by jonringer — APPROVED** [review]
LGTM
shows usage

```
https://github.com/NixOS/nixpkgs/pull/124248

1 package built:
imgcrypt
```

**Pull request merged**